runtime/testy/studentAPI.py

```
@@ -7,16 +7,29 @@ def __init__(self, toManager, fromManager):
     # Wait for ack from SM before continuing
     self.fromManager.recv()
 
+  def getState(self):
+    self.toManager.put([SM_COMMANDS.GET_STATE])
+    return self.fromManager.recv()
+
   def getValue(self, key):
     """Returns the value associated with key
     """
     # TODO: Actually use key
+    if key not in self.getState().keys():
```

For debugging purposes, you can just raise the error without any if statements.